src/client.rs

```diff
@@ -152,9 +172,24 @@ impl Client {
             .get(url)
             .headers(headers)
             .send()
-            .await?
-            .error_for_status()?;
-        Ok(response)
+            .map_err(Error::from);
+
+        Box::pin(response.and_then(move |response| {
+            Box::pin(
+                response
+                .bytes()
```

Comment:
You should be able to use the `json` method here instead of deserializing manually.

Reply:
Nice